python/ray/_private/worker.py

```diff
@@ -1283,6 +1283,9 @@ def init(
 
     if local_mode:
         driver_mode = LOCAL_MODE
+        logger.warning(
+            "DeprecationWarning: local mode is deprecated and will be removed in the future. It causes memory leak and should not be used."
```

Comment:
> It causes memory leak and should not be used.

actually this might not be true, at least for the case we are looking into today. So we can just something this is an experimental feature that's no longer maintained.

Reply:
Also shall we have a consistent style to warn the deprecation?
It looks we have at least two styles: logger.warning() v.s. warning.warn(). And the later might be preferred from https://docs.python.org/2/howto/logging.html#when-to-use-logging, which says:
"[warnings.warn()](https://docs.python.org/2/library/warnings.html#warnings.warn) in library code if the issue is avoidable and the client application should be modified to eliminate the warning"
"[logging.warning()](https://docs.python.org/2/library/logging.html#logging.warning) if there is nothing the client application can do about the situation, but the event should still be noted"

In case of deprecation, clients can actually do something.

Reply:
Sounds good, let me know how this looks now?

Reply:
LG!